### ISSUE TITLE
Binding and unbinding builtins results in incorrect resolution of the identifier

### ISSUE THREAD

**Issue by QuietRocket** [microsoft/pyright#1338]
**Describe the bug**
When a builtin variable is bound and unbound, Pyright (and consequently Pylance) reports that the variable is unbound, while in reality it should still be bound to the builtin.

**To Reproduce**
Bind a builtin function to something, then unbind it with `del`. Once unbound from the local definition, Python will interpret the identifier as its original builtin value, but Pyright flags that the variable is unbound.

**Expected behavior**
Pyright shouldn't flag the variable as unbound, as Python rebinds to the builtin.

**Screenshots or Code**
```python
import builtins

print = builtins.print

print(1)

del print

print(2)
```
<img width="658" alt="Screen Shot 2021-01-05 at 11 24 11 AM" src="https://user-images.githubusercontent.com/56984521/103689823-ab976f00-4f48-11eb-857b-1bd9eab78948.png">

**VS Code extension or command-line**
Pyright 1.1.99

**Additional context**
Running Python 3.9.1

**Comment by erictraut** [comment]
Fixing this would require adding quite a bit of extra complexity, and I don't think it's something that will impact most users. I would recommend against reassigning builtin symbols in the first place, and deleting the local version and expecting the builtin to remain is even stranger. For that reason, I'm going to mark this as "as designed". You can either change your code to not use this pattern or use a "# type: ignore" to suppress the error.
